### PR TITLE
add zlibcomplete package

### PR DESCRIPTION
**Pull request by romch007**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/z/zlibcomplete/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    on_install(function (package)
+        local zlc_headers_dir = path.join(package:installdir("include"), "zlc")
+        os.mkdir(zlc_headers_dir)
+        os.cp("lib/zlc/*.hpp", zlc_headers_dir)
+        local configs = {}
+        table.insert(configs, "-DZLIBCOMPLETE_EXAMPLES=off")
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        -- if package:config("shared") then
+        --   table.insert(configs, "-DZLIBCOMPLETE_SHARED=on")
+        -- else
```
Comment (waruqi) [inline packages/z/zlibcomplete/xmake.lua:21]: You need to distinguish between static/shared, and remove unused comments.

File: packages/z/zlibcomplete/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    add_urls("https://github.com/rudi-cilibrasi/zlibcomplete/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/rudi-cilibrasi/zlibcomplete.git")
+    add_versions("1.0.5", "2b263983823395eaabb091cb9b629eb8466fe24b929bba7ff6d833cadad11977")
+
+    add_deps("cmake", "zlib")
+
+    on_install(function (package)
+        local zlc_headers_dir = path.join(package:installdir("include"), "zlc")
+        os.mkdir(zlc_headers_dir)
+        os.cp("lib/zlc/*.hpp", zlc_headers_dir)
```
Comment (waruqi) [inline packages/z/zlibcomplete/xmake.lua:15]: we only need 

```console
        import("package.tools.cmake").install(package, configs)
        os.cp("lib/zlc/*.hpp", package:installdir("include", "zlc"))
```

and move it to tail.

File: packages/z/zlibcomplete/xmake.lua (hunk 1 above)
Comment (romch007) [reply packages/z/zlibcomplete/xmake.lua:18]: I can't make shared build mode work for know. How can I remove the option ?

Comment (waruqi) [reply packages/z/zlibcomplete/xmake.lua:18]: see https://github.com/xmake-io/xmake-repo/blob/1c759cf020e6e7128482f11f11a4876270b610c5/packages/s/spdlog/xmake.lua#L24

**Pull request merged**